### ISSUE TITLE
Testing OPTIONS Pre-flight request with Codeception 

### ISSUE THREAD

**Issue by AdrianSkierniewski** [barryvdh/laravel-cors#159]
Hi,

First of all, I'd like to thank you for such a great package!
I have some problems with this condition in CORS service provider

```PHP
if ($request->isMethod('OPTIONS')) {
         $kernel->prependMiddleware(HandlePreflight::class);
   }
```
The problem is related to the way how CORS is adding this middleware. At the bootstrap stage, it's often different than later during tests. Removing that condition fixes the problem because kernel always has access to correct $request object.

So my question is, would it be ok to remove this condition from a service provider and just add that middleware always?

**Comment by AdrianSkierniewski** [comment]
@barryvdh I can create PR with this fix if you're ok with it

**Comment by barryvdh** [comment]
Closing this as part of an issue-cleanup. Please re-open if this is still an issue on the latest version.

**Comment by AdrianSkierniewski** [comment]
@barryvdh I confirmed that latest version works :) but I notice that now we're returning 405 Method Not Allowed without CORS headers. Is this expected behavior? I can easily find use case when someone used a wrong method in ajax call and instead Method Not Allowed error he sees CORS error in a web browser, so it can be confusing. I changed my assertions in tests to match new behavior but I'd like to ask about your opinion about this one too.
